### PR TITLE
feat(hud): resolve active plugin root via OMC_PLUGIN_ROOT env var

### PR DESCRIPTION
**Pull request by pgagarinov**
## Summary

Fixes a bug where the HUD statusline loads a stale version of the OMC bundle when Claude Code is launched with `--plugin-dir <local-repo>`. The wrapper currently resolves the HUD bundle via a hardcoded `OMC_DEV=1` + three-fixed-paths fallback (`~/Workspace`, `~/workspace`, `~/projects`) — any other checkout location (e.g. `~/_Git`, `~/code`, `~/src`) lands in the stale marketplace clone at the previously-installed version instead of the freshly-built local one.

The fix is a per-process env-var handshake between the `omc` CLI launcher and the HUD wrapper, plus a matching `--plugin-dir-mode` flag on `omc setup` so the installer doesn't duplicate skills/agents the plugin already provides at runtime.

## Why one PR, not three

I considered splitting into env-var / setup-flag / refactor. They share a single root cause and the tests straddle the boundaries, so splitting produces broken intermediate states:

- The CLI argv capture without the wrapper reader is an env var nobody reads.
- The wrapper reader without the CLI capture requires users to manually export `OMC_PLUGIN_ROOT` (acceptable as a workaround, not as a product).
- The `--plugin-dir-mode` flag is useless if the HUD bundle still fails to resolve under `--plugin-dir`.
- The template extraction and constants centralisation exist **because** of the new test coverage — the drift-guard and consistency tests have nothing to guard without them.

Happy to split if preferred, but flagging the cost up front. This PR is backend plumbing only, no UX surface, no preset changes.

## What's out of scope

- **No `dist/` or `bridge/` churn.** Source-only diff.
- **No changes to HUD default presets or element placement.** Zero user-visible behavior change unless `--plugin-dir` is actually used.
- **No new config file writes.** Explicitly rejected persistence to `.omc-config.json` because it goes stale across alias re-points and collides across parallel `CLAUDE_CONFIG_DIR` profiles sharing one checkout.

## Root cause

Claude Code does not propagate `--plugin-dir` to the statusline subprocess — neither via env var nor via the `StatuslineStdin` JSON payload. So even when the user passes `--plugin-dir <local-repo>`, the HUD wrapper has no runtime signal about where the active plugin lives, and it falls through to filesystem heuristics that pick the wrong version.

## Fix

**1. CLI argv capture (`src/cli/launch.ts`)**
The `omc` CLI parses `--plugin-dir <path>` from its own argv and sets `OMC_PLUGIN_ROOT=<abs path>` in the spawned Claude Code process env. The flag is still forwarded to Claude Code untouched — the capture is purely additive. Also added to `TMUX_ENV_FORWARD` so tmux-launched sessions inherit it.

**2. Wrapper resolution order (`scripts/lib/hud-wrapper-template.txt`)**
New highest-priority resolution step: if `process.env.OMC_PLUGIN_ROOT` is set and `<root>/dist/hud/index.js` exists, import from there. Existing cache / marketplace / npm fallbacks remain for all other launch modes. The arbitrary `OMC_DEV=1` + hardcoded-path branch is deleted — `OMC_PLUGIN_ROOT` subsumes it cleanly and works for any checkout location.

**3. `omc setup --plugin-dir-mode` flag (`src/cli/index.ts`, `src/installer/index.ts`)**
New flag (auto-enabled when `OMC_PLUGIN_ROOT` is set at setup time) that skips copying agents and bundled skills into `<configDir>`, since the plugin provides them at runtime via `--plugin-dir`. HUD wrapper, hooks, CLAUDE.md merge, and `.omc-config.json` are still installed. Conflict with `--no-plugin` resolved in favor of `--no-plugin` with a warning.

**4. Deduplication**
The HUD wrapper body was previously duplicated inline in `src/installer/index.ts` (~186 lines) and `scripts/plugin-setup.mjs` (~186 lines). Extracted to `scripts/lib/hud-wrapper-template.txt` as the single source of truth, consumed via `buildHudWrapper()` from `src/lib/hud-wrapper-template.ts` (TS) and a thin mirror at `scripts/lib/hud-wrapper-template.mjs` (for the ESM install path). A new drift-guard test asserts both mirrors read the same file.

**5. Constants centralisation (`src/lib/env-vars.ts`, `src/lib/paths.ts`)**
Magic strings like `OMC_PLUGIN_ROOT`, `plugins/cache/omc/oh-my-claudecode`, `dist/hud/index.js`, `hud/omc-hud.mjs`, and `.omc-config.json` were scattered across the TS source. Centralised into two small modules with a consistency test that asserts the non-TS template files stay in sync with the TS constants.

## Parallel installs

The env-var approach supports parallel `--plugin-dir` installs of different versions because the discriminator lives in the process env, not on disk. Two terminals launched with different `--plugin-dir` values each load their own version, even when sharing the same `CLAUDE_CONFIG_DIR`. This was the motivation for rejecting the "persist pluginRoot to .omc-config.json" alternative.

## Tests (+14)

- **`src/cli/__tests__/plugin-dir-capture.test.ts`** — module-level `vi.mock` of `execFileSync` captures the env passed to spawned Claude Code. 6 cases covering `--plugin-dir X`, `--plugin-dir=X`, absent, parent-env-only, argv-wins-over-parent-env, relative-path-resolved-to-absolute.
- **`src/cli/__tests__/setup-command-precedence.test.ts`** — drives the real commander program via the new `buildProgram()` export; mocks `install` to capture `InstallOptions`. 6 cases covering flag, env-var-auto-detect, neither, `--no-plugin` conflict, combined `--force`.
- **`src/installer/__tests__/plugin-dir-mode-e2e.test.ts`** — calls `install()` against a tmpdir `CLAUDE_CONFIG_DIR`; asserts filesystem shape (no `agents/`, no `skills/`; `hud/`, `CLAUDE.md`, `settings.json` present).
- **`src/installer/__tests__/hud-wrapper-env.test.ts`** — spawns the generated wrapper as a child process with scrubbed `PATH` / `NODE_PATH` to defeat globally-installed npm fallbacks; pins the specific fallback step that fires via a unique `CACHE_STUB_MARKER` for each of the 5 resolution cases.
- **`src/__tests__/hud-wrapper-template-sync.test.ts`** — drift guard: both TS and MJS template readers must return byte-identical strings.
- **`src/__tests__/paths-consistency.test.ts`** — asserts every path constant from `src/lib/paths.ts` appears literally in `scripts/lib/hud-wrapper-template.txt` where expected, so the non-TS template can't drift from the TS constants.

Full suite on this branch: the 40 new/adjacent tests all pass. Pre-existing macOS symlink test failures present on upstream/dev (unrelated to this PR) remain unchanged.

## Size

- 23 files changed
- ~1350 additions, ~460 deletions
- Net line count shrinks in the installer due to the template extraction; the remaining growth is in focused tests.
- Source-only. No `dist/`, no `bridge/`.

## Test plan

- [ ] `npm test` on a Linux machine (pre-existing macOS symlink failures do not apply to upstream CI)
- [ ] Live: launch Claude Code via `omc --plugin-dir <local-repo>`, verify HUD reports the local-repo version
- [ ] Regression: launch Claude Code in marketplace/plugin-cache mode, verify HUD still resolves correctly (env var not set → existing fallbacks fire)
- [ ] `omc setup --plugin-dir-mode` → no agents/ or skills/ written under target `CLAUDE_CONFIG_DIR`
- [ ] `omc setup` with `OMC_PLUGIN_ROOT` exported → same behavior (auto-detected)
- [ ] `omc setup --plugin-dir-mode --no-plugin` → warns and honors `--no-plugin`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/lib/hud-wrapper-template.txt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  // 2. Plugin cache (for production installs)
+  // Respect CLAUDE_CONFIG_DIR so installs under a custom config dir are found
+  const configDir = getClaudeConfigDir();
+  const pluginCacheBase = join(configDir, "plugins", "cache", "omc", "oh-my-claudecode");
+  if (existsSync(pluginCacheBase)) {
+    try {
+      const versions = readdirSync(pluginCacheBase);
+      if (versions.length > 0) {
+        const sortedVersions = versions.sort((a, b) => a.localeCompare(b, undefined, { numeric: true })).reverse();
```
[the remark this answers is not in the document]
Comment (pgagarinov) [reply scripts/lib/hud-wrapper-template.txt]: Thanks — fixed in 3196784b. One clarification: this isn't a regression from the template extraction. Both the pre-PR inline wrapper in `src/installer/index.ts` and the one in `scripts/plugin-setup.mjs` used the same `localeCompare(..., { numeric: true })` sort; there was no prerelease-handling branch to lose. So this is a latent pre-existing bug that I'm now correcting since we're already touching this code.

Fix: replaced the sort with a semver-aware descending comparator that parses `[major.minor.patch]-<prerelease>`, compares the numeric triple first, and treats "no prerelease tag" as greater than any prerelease tag for the same `[M.m.p]`. So `1.0.1` now wins over `1.0.1-alpha`, `1.0.0-rc.1` loses to `1.0.0`, etc.

Added regression test `case 6: cache step is semver-aware — stable beats prerelease with same [M.m.p]` in `src/installer/__tests__/hud-wrapper-env.test.ts` that stubs two cache versions (`1.0.0-alpha` and `1.0.0`) with distinct stdout markers and asserts the stable one is loaded. A naive `localeCompare(numeric)` sort would fail this test.

**Comment by Yeachan-Heo** [comment]
CI is red on a HUD wrapper regression test (`src/__tests__/hud-windows.test.ts`) after the wrapper/template change. I'm taking this into an immediate follow-up fix path now.

—
*[repo owner's gaebal-gajae (clawdbot) 🦞]*

File: scripts/lib/hud-wrapper-template.txt [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            if (ai === undefined) return 1;
+            if (bi === undefined) return -1;
+            if (typeof ai === "number" && typeof bi === "number") return bi - ai;
+            return String(bi).localeCompare(String(ai), undefined, { numeric: true });
+          }
+          // Same [M.m.p]: stable (empty pre) wins over any prerelease.
+          if (pa.pre === pb.pre) return 0;
+          if (pa.pre === "") return -1;
+          if (pb.pre === "") return 1;
+          return pb.pre.localeCompare(pa.pre);
```
[the remark this answers is not in the document]
Comment (pgagarinov) [reply scripts/lib/hud-wrapper-template.txt]: This is pre-existing

[the remark this answers is not in the document]
Comment (pgagarinov) [reply scripts/lib/hud-wrapper-template.txt]: Fixed in fc72b2f3. Added `{ numeric: true }` to the prerelease-tiebreak `localeCompare` so `rc.10` now sorts above `rc.2`. New regression test `case 7: cache step orders prerelease tags numerically — rc.10 beats rc.2` in `src/installer/__tests__/hud-wrapper-env.test.ts` stubs two cache directories (`1.0.0-rc.2` and `1.0.0-rc.10`) with distinct stdout markers and asserts `rc.10` is loaded. Thanks — this was a real edge case my initial fix missed.
